tests/integration/feature_repos/test_repo_configuration.py

```diff
@@ -0,0 +1,343 @@
+import tempfile
+import uuid
+from contextlib import contextmanager
+from dataclasses import dataclass, replace
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Dict, List, Optional, Union
+
+import pytest
+
+from feast import FeatureStore, FeatureView, RepoConfig, driver_test_data, importer
+from feast.data_source import DataSource
+from tests.data.data_creator import create_dataset
+from tests.integration.feature_repos.universal.data_source_creator import (
+    DataSourceCreator,
+)
+from tests.integration.feature_repos.universal.entities import customer, driver
+from tests.integration.feature_repos.universal.feature_views import (
+    create_customer_daily_profile_feature_view,
+    create_driver_hourly_stats_feature_view,
+)
+
+
+@dataclass(frozen=True, repr=True)
+class TestRepoConfig:
+    """
+    This class should hold all possible parameters that may need to be varied by individual tests.
+    """
+
+    provider: str = "local"
+    online_store: Union[str, Dict] = "sqlite"
+
+    offline_store_creator: str = "tests.integration.feature_repos.universal.data_sources.file.FileDataSourceCreator"
+
+    full_feature_names: bool = True
+    infer_event_timestamp_col: bool = True
+
+
+def ds_creator_path(cls: str):
+    return f"tests.integration.feature_repos.universal.data_sources.{cls}"
+
+
+DYNAMO_CONFIG = {"type": "dynamodb", "region": "us-west-2"}
+REDIS_CONFIG = {"type": "redis", "connection_string": "localhost:6379,db=0"}
+FULL_REPO_CONFIGS: List[TestRepoConfig] = [
+    # Local configurations
+    TestRepoConfig(),
+    TestRepoConfig(online_store=REDIS_CONFIG),
+    # GCP configurations
+    TestRepoConfig(
+        provider="gcp",
+        offline_store_creator=ds_creator_path("bigquery.BigQueryDataSourceCreator"),
+        online_store="datastore",
+    ),
+    TestRepoConfig(
+        provider="gcp",
+        offline_store_creator=ds_creator_path("bigquery.BigQueryDataSourceCreator"),
+        online_store=REDIS_CONFIG,
+    ),
+    # AWS configurations
+    TestRepoConfig(
+        provider="aws",
+        offline_store_creator=ds_creator_path("redshift.RedshiftDataSourceCreator"),
+        online_store=DYNAMO_CONFIG,
+    ),
+    TestRepoConfig(
+        provider="aws",
+        offline_store_creator=ds_creator_path("redshift.RedshiftDataSourceCreator"),
+        online_store=REDIS_CONFIG,
+    ),
+]
+
+
+OFFLINE_STORES: List[str] = []
+ONLINE_STORES: List[str] = []
+PROVIDERS: List[str] = []
+
+
+@dataclass
+class Environment:
+    name: str
+    test_repo_config: TestRepoConfig
+    feature_store: FeatureStore
+    data_source: DataSource
+    data_source_creator: DataSourceCreator
+
+    end_date = datetime.now().replace(microsecond=0, second=0, minute=0)
```

Comment:
How do we decide between whether to put configuration in the `Environment` object vs `TestRepoConfig`. At first glance it seems like these dates could be a desired state that the user can configure through the `TestRepoConfig`